doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,390 @@
+|0.3| Revision in Title,Switchport modes and CLI enhancements| [Muhammad Hamza Iqbal](https://github.com/ham-xa)| [Hafiz Mati ur Rehman](https://github.com/Mati86) , [Rida Hanif](https://github.com/ridahanif96) , [Umar Asad](https://github.com/MuhammadUmarAsad) & [Arsalan Ahmad](https://github.com/ahmadarsalan/)| 24 Oct 2022|
+  
+## Table of Contents
+
+- Scope  
+- Definitions/Abbreviations  
+- Overview  
+- Introduction
+- Topology Design 
+- Architecture Design  
+- High-level Design  
+  - Switchport Mode  
+  - Add Multiple Vlan(s) 
+  - Delete Multiple Vlan(s) 
+  - Add Multiple Vlan(s) on Trunk Ports  
+  - Delete Multiple Vlan(s) on Trunk Ports   
+- CLI & YANG Model Configuration Commands & Usage  
+  - CLI Configuration Commands 
+  - YANG Model Configurations 
+  - Example/Usage of Commands 
+- SAI API 
+- Warm Boot and Fastboot Design Impact  
+- Restrictions/Limitations 
+- Testing Requirements/Design  
+    - Unit Test cases  
+- Future Work
+
+## Scope
+
+This high-level design document describes the implementation of switchport modes and CLI enhancements in SONiC.
+
+## Definitions/Abbreviations
+
+| **Sr No** |  **Term**   |  **Definition**                      |
+| :------------- |:-------------| :-----|
+|  1       |  VLAN_ID      |  Unique identifier for each VLAN     |  
+|  2       |  Dot1q        |  IEEE 802.1Q Protocol                |  
+|  3       |  Trunk  |  The port is untagged member to one VLAN and tagged member to one or more VLANs  |  
+|  4       | Access |  The port is untagged member to only one VLAN|  
+|  5       |  Routed | The port is in L3 interface mode                       |  
+|  6       |  Hybrid|   The port is multiple untagged and multiple tagged vlan members to  more VLANs         |  
+
+
+## Overview
+
+This HLD will provide implementation details of switch port modes access and trunk on a Port or on a PortChannel. In this HLD vlan configuration commands are also improved for efficient configuration and management of VLANs.
+
+## Introduction
+
+Switch ports are Layer 2-only interfaces associated with a physical port. A switch port can have three modes: access port, a trunk port, or a hybrid port. We can manually configure a port as an access port, trunk port or hybrid port depending on requirement . Switch ports are used for managing the physical interface and associated Layer 2 protocols and do not handle routing.
+
+### Access Mode
+
+In access mode a port possesses the following properties
+* Receives untagged/tagged traffic for a particular Vlan
+* Forwards untagged/tagged traffic for a particular Vlan
+
+### Trunk Mode
+
+In trunk mode a port possesses the following properties
+* Receives untagged traffic for a single Vlan
+* Forwards untagged traffic for a single Vlan
+* Receives tagged traffic for one or more Vlan(s)
+* Forwards tagged traffic for one or more Vlan(s)
+
+### Hybrid Mode
+
+In hybrid mode a port possesses the following properties
```

Comment:
Is this planned to be added as a CLI option? I don't see it in example below

Reply:
Hybrid mode is the future scope of this HLD. This HLD has two switchport modes (access, trunk). 
For hybrid mode, we have a separate [HLD](https://github.com/sonic-net/SONiC/issues/1136) which is in the roadmap of 202305 and we will present it later in one of the future community reviews.